### ISSUE TITLE
[Bugs] Plugin upgrade reported as success, but in reality it fails

### ISSUE THREAD

**Issue by mystralkk** [Geeklog-Core/geeklog#494]
**Reported by Roccivic on 17 May 2011 12:21**

**Version:** 1.8.0b2

[Mantis Bugtracker #1345](http://project.geeklog.net/tracking/view.php?id=1345)

**Comment by mystralkk** [comment]
**Comment posted by dhaun on 21 May 2011 14:13**

I can confirm the described behaviour. I can also confirm, however, that PLG_upgrade(searchrank) is called. Digging deeper ...

**Comment by mystralkk** [comment]
**Comment posted by dhaun on 21 May 2011 14:39**

It&#039;s pretty simple, actually: When we proceed to unpack the new version, we have already loaded the old functions.inc and we never- during the upgrade - load the new functions.inc. So when we call PLG_upgrade, it goes to the old functions.inc (which is in memory). And in this case, the old functions.inc doesn&#039;t even have an upgrade function to call (since it&#039;s for the very first version of the plugin).

The only thing I don&#039;t understand is how the pi_version gets updated in the database.
